libs/api-services/src/lib/session/extendSession.ts

```diff
@@ -0,0 +1,49 @@
+import { prisma } from '@dnd-assistant/prisma';
+import { generateSessionExpiry } from './generate-session-expiry';
+import { Session } from '@prisma/client';
+
+/**
+ * We will only extend a session if it expires within this many days
+ */
+const EXTEND_IF_EXPIRES_WITHIN_DAYS = 7;
+/**
+ * When extending a session, it will be valid for this length
+ */
+const EXTENDED_SESSION_VALID_DAYS = 60;
+
+/**
+ * Attempts to extend session if possible, otherwise returns early
+ * Only throws on a DB error
+ */
+export const extendSession = async (session: Session) => {
+  if (new Date(session.expiresAt) < new Date()) {
+    // Session has already expired, cannot be extended
+    return;
+  }
+
+  const extensionWindow = new Date();
+  extensionWindow.setDate(new Date().getDate() + EXTEND_IF_EXPIRES_WITHIN_DAYS);
+  if (new Date(session.expiresAt) > extensionWindow) {
+    // Session does not need to be extended, still has more than EXTEND_IF_EXPIRES_WITHIN_DAYS
+    return;
+  }
+
```

Comment:
don't we still need a usecase here for;
```
if (new Date(session.expiresAt) < extensionWindow && extensionWindow < session.extendableUntil) {
```
so that we can update the session extension by the window periods instead of placing it at the maximum allowed extension?

Reply:
I believe this case is already covered after re-reading, but if you wouldn't mind rephrasing the problem you've noticed that'd help me